### DISCUSSION TITLE
How to get graphiql ide while dependancy Injection in graphql fastapi?

### DISCUSSION THREAD

**Discussion by silicology1** [fastapi/fastapi#7088] [Questions]
This is the code:
```
from starlette.graphql import GraphQLApp
from starlette.requests import Request as Rq
from fastapi_jwt_auth import AuthJWT
    
    
graphql_app = GraphQLApp(schema=graphene.Schema(query=Query, mutation=Mutation))

@router.post("/gql")
async def graph(request: Rq, Authorize: AuthJWT = Depends() ):
    request.state.authorize = Authorize
    return await graphql_app.handle_graphql(request=request)

app.include_router(router)
```

Its working fine with post request on insomnia ide but I am unable to see the graphiql ide in the browser in "localhost:8000/gql" url
It gives error: "GET /gql HTTP/1.1" 405 Method Not Allowed

How can I get the graphiql ide in browser?

Is it possible to get the ide by Custom Request and APIRoute class? https://fastapi.tiangolo.com/advanced/custom-request-and-route/

https://stackoverflow.com/questions/65175492/how-to-get-graphiql-ide-while-dependancy-injection-in-graphql-fastapi

**Comment by ycd** [comment]
@amiyatulu looks like your issue has been resolved in [fastapi-jwt-auth](https://github.com/IndominusByte/fastapi-jwt-auth/issues/28) or are you still looking for an answer?

**Comment by silicology1** [comment]
Yes, closing the issue. Thank you.

**Comment by tiangolo** [comment]
Thanks for the help here @ycd ! :clap: :bow: 

Thanks for reporting back and closing the issue @amiyatulu  :+1:
